libs/blocks/graybox-promote/components/graybox-promote-lit.js

```diff
@@ -114,7 +113,19 @@ const getSharepointData = async (url) => {
   };
 };
 
-const getProjectExcelPath = (url) => url.pathname.replace('.json', '.xlsx');
+const getFilePath = async (ref, repo, owner, excelRef) => {
+  const status = await fetch(`${ADMIN}/status/${owner}/${repo}/${ref}?editUrl=${excelRef}`);
+  const statusJson = await status.json();
+  return (new URL(statusJson?.preview?.url)).pathname;
+};
+
+const preview = async (owner, repo, ref, path) => {
+  const previewResp = await fetch(
+    `${ADMIN}/preview/${owner}/${repo}/${ref}${path}`,
+    { method: 'POST' },
+  );
+  return previewResp.json();
+};
```

Comment:
I'd suggest using the implied return of the arrow function and simply return the promise which is awaited anyway.
```suggestion
const preview = ({owner, repo, ref, path}) => fetch(
    `${ADMIN}/preview/${owner}/${repo}/${ref}${path}`,
    { method: 'POST' },
  ).then(res => res.json())
```

Or one step further
```suggestion
const previewUrl = ({owner, repo, ref, path}) => fetch(
    `${ADMIN}/preview/${owner}/${repo}/${ref}${path}`,
    { method: 'POST' },
  )
  .then(res => res.json())
  .then(preview => new URL(preview.url))
```

Reply:
The desired output is the preview url, so I'll take your suggested shorter version 👍